pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
```

Comment:
Is there value in extremely large endpoint lists?  Is there an upper bound beyond which the service is an antipattern, or should be a different concept?  Is there any value in a pod being able to reach 10k other things?  After all, DNS is a very similar problem and a DNS address with thousands of endpoints would considered a bad thing?  Does very large numbers of endpoint simply mean someone should be doing something differentv


> On Feb 13, 2015, at 10:18 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > @@ -738,7 +771,35 @@ type Endpoints struct {
> >  	TypeMeta   `json:",inline"`
> >  	ObjectMeta `json:"metadata,omitempty"`
> >  
> > -	Endpoints []string `json:"endpoints,omitempty"`
> > +	Endpoints []Endpoint `json:"endpoints,omitempty"`
> We're going to need to support pagination (#2349) and incremental updates, for both writes and reads, for large sets. A solution would be to make /endpoints a list of lists, which naively would make initial creation require a lot more requests, but which could be mitigated by supporting Batches of requests or POSTs of lists.
> 
> —
> Reply to this email directly or view it on GitHub.
>